File: backend/src/models/Restaurant.js
/**
 * Restaurant Model - For multi-restaurant support
 */
const mongoose = require('mongoose');

const restaurantSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true },
  description: { type: String, default: '' },
  image: { type: String, default: '' },
  address: { type: String, required: true },
  phone: { type: String, required: true },
  email: { type: String, required: true },
  owner: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  rating: { type: Number, default: 0, min: 0, max: 5 },
  totalRatings: { type: Number, default: 0 },
  cuisine: [{ type: String }],
  openingHours: {
    open: { type: String, default: '09:00' },
    close: { type: String, default: '22:00' }
  },
  deliveryFee: { type: Number, default: 2.99 },
  minOrder: { type: Number, default: 10 },
  isActive: { type: Boolean, default: true },
  isFeatured: { type: Boolean, default: false }
}, { timestamps: true });

module.exports = mongoose.model('Restaurant', restaurantSchema);

File: frontend/lib/data/models/restaurant.dart
class Restaurant {
  final String id;
  final String name;
  final String description;
  final String image;
  final String address;
  final String phone;
  final double rating;
  final int totalRatings;
  final List<String> cuisine;
  final double deliveryFee;
  final double minOrder;
  final bool isActive;
  final bool isFeatured;
  final String openingHours;

  Restaurant({
    required this.id,
    required this.name,
    this.description = '',
    this.image = '',
    required this.address,
    this.phone = '',
    this.rating = 0,
    this.totalRatings = 0,
    this.cuisine = const [],
    this.deliveryFee = 2.99,
    this.minOrder = 10,
    this.isActive = true,
    this.isFeatured = false,
    this.openingHours = '09:00 - 22:00',
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      image: json['image'] ?? '',
      address: json['address'] ?? '',
      phone: json['phone'] ?? '',
      rating: (json['rating'] ?? 0).toDouble(),
      totalRatings: json['totalRatings'] ?? 0,
      cuisine: List<String>.from(json['cuisine'] ?? []),
      deliveryFee: (json['deliveryFee'] ?? 2.99).toDouble(),
      minOrder: (json['minOrder'] ?? 10).toDouble(),
      isActive: json['isActive'] ?? true,
      isFeatured: json['isFeatured'] ?? false,
      openingHours:
          '${json['openingHours']?['open'] ?? '09:00'} - ${json['openingHours']?['close'] ?? '22:00'}',
    );
  }
}
